tests/sgdr_test.cpp

```diff
@@ -39,12 +39,12 @@ TEST_CASE("SGDRCyclicalResetTest","[SGDRTest]")
       arma::Col<size_t> nextRestart(1000 / 10 /  mult);
       nextRestart(0) = restart;
       for (size_t j = 1; j < nextRestart.n_elem; ++j)
-        nextRestart(j) = nextRestart(j - 1) * mult;
+        nextRestart(j) = nextRestart(j - 1) + restart * std::pow(mult, j);
```

Comment:
A justification here would also be nice.  Sorry for being picky but I am a little paranoid of changing tests that seemed to work before.

Reply:
This test is changed to match the fixes to the method , for example , as above , if T0 is 10 and mult is 2 then , next restart will will take values 10, 20, 40, 80 , not at all what the paper intented , the intented result should be 10, 30, 70, 150 .
My method , on the other hand calculates this , it takes the orginal result values, multiplies it by the power of mult , and then adds it . 
I hope this clarifies my changes .

Reply:
Thanks for the clarification.  The changes make more sense now.  But I disagree with the restart calculations.  The paper reads this:

```
In order to improve anytime performance, we suggest an option to start
with an initially small T_i and increase it by a factor of T_mult at every restart.
```

To me, that means that the number of epochs before restart should be 10, 20, 40, 80 when T_0 is 10 and mult is 2, not 10, 30, 70, 150.

Also, it seems to me like there has not been any change in `CyclicalDecay` to produce the results 10, 30, 70, 150; it multiplies by `multFactor` every time a restart is done.  So honestly I am a bit confused about why this test passes, and I wonder if something is wrong with the test.

Reply:
We should address this comment before merge.  I still don't have a good understanding of what is going on here---please correct me if I've misunderstood anything.